plaso/parsers/viminfo.py

```diff
@@ -0,0 +1,527 @@
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_REGISTERS_ITEM))
+
+  _FILEMARKS_ITEM = pyparsing.Group(
+      pyparsing.Suppress(pyparsing.LineStart()) +  pyparsing.Literal('\'') +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.restOfLine +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      _FILEMARK_ITEM).setResultsName('filemarks_items*')
+
+  _FILEMARKS_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# File marks:') +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_FILEMARKS_ITEM))
+
+  _JUMPLIST_ITEM = pyparsing.Group(
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Word('-\'') +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.restOfLine +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      _FILEMARK_ITEM).setResultsName('jumplist_items*')
+
+  _JUMPLIST_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# Jumplist (newest first):') +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_JUMPLIST_ITEM))
+
+  _HISTORY_MARKS_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# History of marks within files (newest to oldest):') +
+      pyparsing.Suppress(pyparsing.LineEnd()))
+
+  LINE_STRUCTURES = [
+      ('preamble', _PREAMBLE),
+      ('command_line_history', _COMMAND_LINE_HISTORY),
+      ('hlsearch', _HLSEARCH),
+      ('search_pattern', _SEARCH_PATTERN),
+      ('substitute_search', _SUBSTITUTE_SEARCH_PATTERN),
+      ('substitute_string', _SUBSTITUTE_STRING),
+      ('search_string_history', _SEARCH_STRING_HISTORY),
+      ('expression_history', _EXPRESSION_HISTORY),
+      ('input_line_history', _INPUT_LINE_HISTORY),
+      ('debug_line_history', _DEBUG_LINE_HISTORY),
+      ('registers_history', _REGISTERS_HISTORY),
+      ('filemarks_history', _FILEMARKS_HISTORY),
+      ('jumplist_history', _JUMPLIST_HISTORY),
+      ('history_marks_history', _HISTORY_MARKS_HISTORY)]
+
+  _SUPPORTED_KEYS = frozenset([key for key, _ in LINE_STRUCTURES])
+
+  def _ParseCommandLineHistory(self, parser_mediator, structure):
+    """Parses command line history items and creates VimInfoEventData objects.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          lines of text that match a command line history
+    """
+    for index, item in enumerate(structure.get('command_line_items', [])):
+      event_data = VimInfoEventData()
+      event_data.history_value = item[1]
+      event_data.history_type = 'Command Line History'
+      event_data.item_number = index
+
+      timestamp = int(item[5])
```

Comment:
Maybe use PyParseIntCast https://github.com/log2timeline/plaso/blob/main/plaso/parsers/text_parser.py#L38

otherwise catch ValueError and add if this should be a decimal (10) or hexadecimal (16)

repeat this for other locations as well.

Reply:
Updated to use PyParseIntCast (via PyparsingConstants.INTEGER)